inc/dbmysql.class.php

```diff
@@ -1686,4 +1678,73 @@ public static function isNameQuoted($value): bool {
       $quote = static::getQuoteNameChar();
       return is_string($value) && trim($value, $quote) != $value;
    }
+
+   /**
+    * Remove SQL comments
+    * © 2011 PHPBB Group
+    *
+    * @param string $output SQL statements
+    *
+    * @return string
+    */
+   public function removeSqlComments($output) {
+      $lines = explode("\n", $output);
+      $output = "";
+
+      // try to keep mem. use down
+      $linecount = count($lines);
+
+      $in_comment = false;
+      for ($i = 0; $i < $linecount; $i++) {
+         if (preg_match("/^\/\*/", $lines[$i])) {
+            $in_comment = true;
+         }
+
+         if (!$in_comment) {
+            $output .= $lines[$i] . "\n";
+         }
+
+         if (preg_match("/\*\/$/", preg_quote($lines[$i]))) {
+            $in_comment = false;
+         }
+      }
+
+      unset($lines);
+      return trim($output);
+   }
+
+   /**
+    * Remove remarks and comments from SQL
+    * @see DBmysql::removeSqlComments()
+    * © 2011 PHPBB Group
+    *
+    * @param $string $sql SQL statements
+    *
+    * @return string
+    */
+   public function removeSqlRemarks($sql) {
+      $lines = explode("\n", $sql);
+
+      // try to keep mem. use down
+      $sql = "";
+
+      $linecount = count($lines);
+      $output = "";
+
+      for ($i = 0; $i < $linecount; $i++) {
+         if (($i != ($linecount - 1)) || (strlen($lines[$i]) > 0)) {
+            if (isset($lines[$i][0])) {
+               if ($lines[$i][0] != "#" && substr($lines[$i], 0, 2) != "--") {
```

Comment:
Maybe tests should be done on `ltrim($lines[$i])` to handle comments preceded by a blank space.

Reply:
same